scheduler/scheduler_system_test.go

```diff
@@ -3084,3 +3086,63 @@ func TestSystemSched_CSITopology(t *testing.T) {
 	must.Eq(t, structs.EvalStatusComplete, h.Evals[0].Status)
 
 }
+
+func TestSystemSched_OverlappingAllocations(t *testing.T) {
+	ci.Parallel(t)
+	h := NewHarness(t)
+
+	node := mock.Node()
+	node.Name = "good-node-1"
+	must.NoError(t, h.State.UpsertNode(structs.MsgTypeTestSetup, h.NextIndex(), node))
+
+	oldJob := mock.SystemJob()
```

Comment:
This is a great test but let's leave some context for future-us to remember why it has a seemingly impossible setup.

```suggestion
	// Create 2 allocs for 2 versions of the same system job on the same node.
	// This should not happen but has been observed, so the scheduler must "fix" this by stopping the old version.
	oldJob := mock.SystemJob()
```